src/main/scala/nl.knaw.dans.easy.sword2/DepositHandler.scala

```diff
@@ -159,13 +196,18 @@ object DepositHandler {
           files.foreach(file => {
             if (!file.isFile)
               throw InvalidDepositException(id, s"Inconsistent dataset: non-file object found: ${file.getName}")
+            checkAvailableDiskspace(file).get
```

Comment:
I'll do a separate PR for this whole `Try` block, because this looks so bad!

Something that is out of reach for putting a comment: in line 219 you're using a `listFiles`, while you have defined a `listFilesSafe` to make sure null-pointers get handled properly. Shouldn't you use the latter there as well? Also look for other uses of `listFiles` in this project while you're at it.